### ISSUE TITLE
`selector-no-union-class-name` rule false positive

### ISSUE THREAD

**Issue by nwoltman** [kristerkari/stylelint-scss#341]
The `selector-no-union-class-name` rule thinks there are errors here:

```scss
.icon {
  &:last-child { // Error on this line
    margin-left: 0.75rem;
  }

  &#divID { // Error on this line
    margin-left: 0.75rem;
  }
}
```

It seems the rule doesn't realize that pseudo-element selectors and ID selectors aren't class name unions.

cc @pipopotamasu

**Comment by kristerkari** [comment]
Thanks! Looks like there are some things to fix.

**Comment by pipopotamasu** [comment]
@kristerkari thx!
